guacamole/src/main/webapp/app/settings/directives/guacSettingsPreferences.js

```diff
@@ -192,6 +192,9 @@ angular.module('settings').directive('guacSettingsPreferences', [function guacSe
                 $scope.canChangePassword = PermissionSet.hasUserPermission(permissions,
                         PermissionSet.ObjectPermissionType.UPDATE, username);
                         
+            })
+            .error(function permissionsFailed(error) {
+                $scope.canChangePassword = false;
```

Comment:
The more I think on this, the more I think that 404 response is indeed the correct response, and that what you have here is the most correct approach. My original thoughts were to modify the REST API to ensure that the authenticated user can always be queried, but if an extension does not expose data at all, even data related to the authenticated user, 404 is really exactly what should be seen. It's the client-side which needs to be modified to properly understand and deal with that response. So ... looks good.

My only concern with these changes as they stand is that absolutely all errors from the REST API here will be interpreted as successful indications that the user lacks permission.

Reply:
Eh ... unless @jmuehlner disagrees, I withdraw my concerns. I think these changes are fine.

There are ways to explicitly check for and handle the error response, as done in other parts of the admin interface, such that the user will see some useful error message describing the failure...  but in this case, I can't think of any such message which would be appropriate or helpful. It's just the password edit screen.

If the rest of the webapp's interface handled such errors and always warned the user when the service is unavailable (for example), that would be another matter, but that's not currently done, and doing so here would thus be out of scope.